homeassistant/components/climate/__init__.py

```diff
@@ -219,6 +237,32 @@ def away_mode_set_service(service):
         descriptions.get(SERVICE_SET_AWAY_MODE),
         schema=SET_AWAY_MODE_SCHEMA)
 
+    def home_mode_set_service(service):
+        """Set home mode on target climate devices."""
+        target_climate = component.extract_from_service(service)
+
+        home_mode = service.data.get(ATTR_HOME_MODE)
+
+        if home_mode is None:
```

Comment:
After I removed the block per your instructions, the test for wrong parameters no longer passes:

self = <tests.components.climate.test_demo.TestDemoClimate testMethod=test_set_home_mode_bad_attr>

    def test_set_home_mode_bad_attr(self):
        """Test setting the home mode without required attribute."""
        state = self.hass.states.get(ENTITY_CLIMATE)
>       self.assertEqual('on', state.attributes.get('home_mode'))
E       AssertionError: 'on' != None

I put the block back in, as it is in all other services in the climate component. If you don't agree, I suggest that there is a greater systematic discussion about what the climate component should do, not just my small additional method.